src/test/jtx/impl/batch.cpp

```diff
@@ -0,0 +1,169 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2024 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx/batch.h>
+#include <test/jtx/utility.h>
+#include <xrpl/protocol/Batch.h>
+#include <xrpl/protocol/HashPrefix.h>
+#include <xrpl/protocol/Sign.h>
+#include <xrpl/protocol/TxFlags.h>
+#include <xrpl/protocol/jss.h>
+#include <optional>
+#include <sstream>
+
+namespace ripple {
+namespace test {
+namespace jtx {
+
+namespace batch {
+
+// Batch.
+Json::Value
+batch(
+    jtx::Account const& account,
+    uint32_t seq,
+    STAmount const& fee,
+    std::uint32_t flags)
+{
+    Json::Value jv;
+    jv[jss::TransactionType] = jss::Batch;
+    jv[jss::Account] = account.human();
+    jv[jss::RawTransactions] = Json::Value{Json::arrayValue};
+    jv[sfTransactionIDs.jsonName] = Json::Value{Json::arrayValue};
+    jv[jss::Sequence] = seq;
+    jv[jss::Flags] = flags;
+    jv[jss::Fee] = to_string(fee);
+    jv[jss::SigningPubKey] = strHex(account.pk());
```

Comment:
Is this needed? Most jtx transactions don't have it.

Reply:
no, its not needed.